### PR TITLE
modules_dart for pure Dart packages; move analyzer code there

### PR DESCRIPTION
**Pull request by yjbanov**
We have Dart code in `angular2` module that ought to be in its own
package. Examples include Dart analysis plugins, and potentially the
transformers (although transformers cannot be moved out just yet).
However, this code is Dart-only and it doesn’t make sense to use JS
directory layout for it. This commit introduces a sub-directory called
`modules_dart`. All modules in this directory are pure Dart packages
using standard pub directory layout. The code in these packages never
gets transpiled. It is directly copied to `dist` unmodified, except an
adjustment in relative paths in `pubspec.yaml` files.

### CODE REVIEW

File: modules_dart/analysis_plugin/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+version: 0.0.0
+description: Dart analyzer plugin for Angular 2
+environment:
+  sdk: '>=1.9.0-dev.8.0'
+dependencies:
+  angular2: '0.0.0'
+  analyzer: '^0.24.4'
+dependency_overrides:
+  angular2:
+    path: ../dist/angular2
```
Comment (tbosch) [inline modules_dart/analysis_plugin/pubspec.yaml]: Should be 2 level up, could you try opening the project in Dart Editor?

File: gulpfile.js [hunk 2]
```diff
@@ -508,9 +508,67 @@ gulp.task('test.transpiler.unittest', function() {
...
+        // dependency_overrides:
+        //   angular2:
+        //     path: ../angular2
+        //
+        var pubspec = yaml.safeLoad(file.contents.toString());
+        var overrides = pubspec['dependency_overrides'];
+        if (overrides) {
+          Object.keys(overrides).forEach(function(pkg) {
+            var overridePath = overrides[pkg]['path'];
+            if (overridePath.startsWith('../dist/')) {
```
Comment (tbosch) [inline gulpfile.js]: Should be 2 up...

File: gulpfile.js [hunk 3]
```diff
@@ -508,9 +508,67 @@ gulp.task('test.transpiler.unittest', function() {
...
+  var through2 = require('through2');
+  var yaml = require('js-yaml');
+
+  return gulp
+    .src([
+      'modules_dart/**/*.dart',
+      'modules_dart/**/pubspec.yaml',
+    ])
+    .pipe(through2.obj(function(file, enc, done) {
+      if (file.path.endsWith('.yaml')) {
```
Comment (tbosch) [inline gulpfile.js]: ends with `pubspec.yaml`

File: gulpfile.js [hunk 4]
```diff
@@ -508,9 +508,67 @@ gulp.task('test.transpiler.unittest', function() {
...
+        //   angular2:
+        //     path: ../angular2
+        //
+        var pubspec = yaml.safeLoad(file.contents.toString());
+        var overrides = pubspec['dependency_overrides'];
+        if (overrides) {
+          Object.keys(overrides).forEach(function(pkg) {
+            var overridePath = overrides[pkg]['path'];
+            if (overridePath.startsWith('../dist/')) {
+              overrides[pkg]['path'] = overridePath.replace('../dist/', '../');
```
Comment (tbosch) [inline gulpfile.js]: Same here...

**Comment by tbosch** [comment]
Please test locally, LGTM.

File: modules_dart/analysis_plugin/pubspec.yaml (hunk 1 above)
Comment (yjbanov) [inline modules_dart/analysis_plugin/pubspec.yaml]: Done

File: gulpfile.js (hunk 2 above)
Comment (yjbanov) [inline gulpfile.js]: Done

**Pull request closed without merging**